packages/walletconnect/src/index.ts

```diff
@@ -156,30 +138,26 @@ export class WalletConnect extends Connector {
 
     const cancelActivation = this.actions.startActivation()
 
-    // if we're trying to connect to a specific chain that we're not already initialized for, we have to re-initialize
-    if (desiredChainId && desiredChainId !== this.provider?.chainId) await this.deactivate()
-
     try {
-      await this.isomorphicInitialize(desiredChainId)
-
-      const accounts = await this.provider
-        ?.request<string[]>({ method: 'eth_requestAccounts' })
-        // if a user triggers the walletconnect modal, closes it, and then tries to connect again,
-        // the modal will not trigger. by deactivating when this happens, we prevent the bug.
-        .catch(async (error: Error) => {
-          if (error?.message === 'User closed modal') await this.deactivate()
-          throw error
-        })
-
-      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-      const chainId = await this.provider!.request<string | number>({ method: 'eth_chainId' }).then((chainId) =>
-        parseChainId(chainId)
-      )
-
-      this.actions.update({ chainId, accounts })
-    } catch (error) {
+      // @todo we need to provide a way to initialise with a given chain. waiting for an update
+      // whether `desiredChainId` is first item in `chains` array 
+      const provider = await this.isomorphicInitialize()
+
+      await provider.enable()
+
+      // if a user triggers the walletconnect modal, closes it, and then tries to connect again,
+      // the modal will not trigger. by deactivating when this happens, we prevent the bug.
+      //
+      // @todo `provider.enable` no longer throws an error when the user closes the modal.
+      // Find a better way.
+      //
+      // .catch(async (error: Error) => {
+      //   if (error?.message === 'User closed modal') await this.deactivate()
```

Comment:
probably goes with "find a better way" but error message strings can be brittle on lib updates - wonder if there is anything you can do there

Reply:
Well, right now, the WC2 team doesn't throw an error at all, I will make sure to give us something like a code at least. From our perspective, good to do in the future, but for now, we throw strings everywhere.